lib/dpl/provider/pypi.rb

```diff
@@ -38,9 +38,30 @@ def pypi_skip_existing_option
           end
       end
 
+      def pypi_pin_setuptools_version
+        setuptools_version = options[:setuptools_version] || context.env['SETUPTOOLS_VERSION'] || ''
```

Comment:
Need some processing of the user-supplied option is necessary. Imagine, for example, a string including `;` is given.

And appropriate specs are required.

Reply:
I have tried to make it possible for users to pin the version they would like either within the `.travis.yml` file or as environment variables. The second option would work better for our use-case, so I wanted to allow both.

WRT user input, I can do a basic checking, but, if the user inputs a weird string, it would fail when trying to install. Of course, this opens an attack surface here, which could allow users to inject things to be pip installed, though.